aep/general/0231/aep.md.j2

```diff
@@ -1,5 +1,168 @@
+  - The field **should** identify the [resource type][aep-122-names] that it
+    references.
+  - The comment for the field **should** document the resource pattern.
+- Other fields besides `name` **may** be "hoisted" from the [standard Get
+  request][request-message]. There is no way to allow for these fields to
+  accept different values for different resources; if this is needed, use the
+  [alternative request form](#nested-request-objects).
+- Batch get **should not** support pagination because transactionality across
+  API calls would be extremely difficult to implement or enforce, and the
+  request defines the exact scope of the response anyway.
+- The request **must not** contain any other required fields, and **should
+  not** contain other optional fields except those described in this or another
+  AEP.
+- The comment above the `names` field **should** document the maximum number of
+  requests allowed.
+
+{% tab oas %}
+
+- The request **must** include a query parameter which accepts an array of
+  resource names specifying the resources to retrieve. The parameter **should**
+  be named `names`.
+  - If no resource names are provided, the API **should** error with status
+    code `400 Bad Request`.
+  - The parameter **should** be required.
+  - The parameter **should** identify the [resource type][aep-122-names] that
+    it references.
+  - The comment for the parameter **should** document the resource pattern.
+- Other parameters besides `name` **may** be "hoisted" from the [standard Get
+  request][request-message]. There is no way to allow for these parameters to
+  accept different values for different resources; if this is needed, use the
+  [alternative request form](#nested-request-objects).
+- The request **must not** contain any other required parameters, and **should
+  not** contain other optional parameters except those described in this or
+  another AEP.
+- The `names` parameter **should** document the maximum number of values
+  allowed in the array.
+
+{% endtabs %}
+
+- Batch get **should not** support pagination because transactionality across
+  API calls would be extremely difficult to implement or enforce, and the
+  request defines the exact scope of the response anyway.
+
+### Response
+
+The response for a batch get method **should** be specified with the following
+pattern:
+
+{% tab proto -%}
+
+```proto
+message BatchGetBooksResponse {
+  // Books requested.
+  repeated Book books = 1;
+}
+```
+
+- The response message **must** include one repeated field corresponding to the
+  resources being retrieved.
+
+{% tab oas %}
+
+- The response schema **must** be an array with each item containing one of the
+  requested resources.
+
+{% endtabs %}
+
+- The order of resources in the response **must** be the same as the names in
+  the request.
+
+## Changelog
+
+- **2024-04-22:** Adopt from Google AIP https://google.aip.dev/231 with the
+  following changes:
+  - Dropped the "nested requests" pattern.
```

Comment:
Why?

Reply:
Because the AIP says "This is generally discouraged unless your use case really requires it." This made it sound like it was added for some strange use case at Google.

Reply:
Also, this is going to be messy in HTTP. Does this still use an HTTP get method? If so, I _hope_ the array of requests is not sent in the body -- HTTP does not allow GET to have a request body. But if not then I suppose they are sent in the query string, and that is really ugly. OAS does allow parameters of type "deep object" but the behavior of these is ill-defined.

Reply:
I think we've settled on not supporting this feature.